tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,157 @@
+             Select container file to be used to build a container image
+             that is then used by bootc image builder to create a disk image.
+
+             Cannot be used with containerimage.
+             """)
+
+    containerfile_workdir: str = field(
+        default=".",
+        option=('--containerfile-workdir'),
+        metavar='CONTAINERFILE_WORKDIR',
+        help="""
+             Select working directory for the podman build invocation.
+             """)
+
+    containerimage: Optional[str] = field(
+        default=None,
+        option=('--containerimage'),
+        metavar='CONTAINERIMAGE',
+        help="""
+             Select container image to be used to build a bootc disk.
+             This takes priority over containerfile.
+             """)
+
+    add_deps: bool = field(
+        default=True,
+        is_flag=True,
+        option=('--add-deps'),
+        help="""
+             Add tmt dependencies to the supplied container image or image built
+             from the supplied Containerfile.
+             This will cause a derived image to be built from the supplied image.
+             """)
+
+    def set_image(self, image: str) -> None:
+        self.image = image
+
+
+@tmt.steps.provides_method('bootc')
+class ProvisionBootc(tmt.steps.provision.ProvisionPlugin[BootcData]):
+    """
+    Provision guest using bootc
+
+    Minimal configuration using the latest nothing image:
+
+    .. code-block:: yaml
+
+        provision:
+            how: bootc
+            containerfile: file://Containerfile
+
+    """
+
+    _data_class = BootcData
+    _guest_class = GuestTestcloud
+    _guest = None
+
+    # build a "modified" container image from the base image with tmt dependencies added
+    def _build_derived_image(self, base_image: str) -> str:
+        self._logger.debug("Building modified container image with necessary tmt packages/config")
+        containerfile_template = Template('''
+            FROM $base_image
+
+            RUN \
+            dnf -y install cloud-init rsync && \
+            ln -s ../cloud-init.target /usr/lib/systemd/system/default.target.wants && \
+            rm /usr/local -rf && ln -sr /var/usrlocal /usr/local && mkdir -p /var/usrlocal/bin
+        ''')
+        containerfile = containerfile_template.substitute({"base_image": base_image})
+        with open(f'{self.workdir}/Containerfile', 'w') as file:
+            file.write(containerfile)
+
+        image_tag = "localhost/tmtmodified"  # TODO: unique ID
+        tmt.utils.Command(
+            "podman", "build", f'{self.workdir}',
+            "-f", f'{self.workdir}/Containerfile',
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+
+        return image_tag
+
+    # build the "base" or user supplied container image
+    def _build_base_image(self, containerfile: str, workdir: str) -> str:
+        image_tag = "localhost/tmtbase"  # TODO: how do I get a unique ID here?
```

Comment:
I think using the tmt run # will work for this. Is there a good way to access it here? Or is there another unique ID I could use for the image tag?

Reply:
How unique does the "unique ID" need to be? Do you wish it to be the same when the user re-runs the previous run, or should it be unique every time `tmt run` is invoked?

Reply:
I think keeping the ID the same when re-running the previous run will work. If I understand how tmt runs work, this will allow modifying the test code without rebuilding the bootc disk which can take a few minutes.